### DISCUSSION TITLE
Will wagmi support ethers@6?

### DISCUSSION THREAD

**Discussion by hammeiam** [wevm/wagmi#2127] [Q&A]
etherjs v6 moves from BigNumber to the native BigInt which is a big usability upgrade. However, wagmi has a peer dep on ethers@5. Is version restriction necessary? Can it be changed?

**Comment by tmm** [comment]
> Is version restriction necessary?

Yes, there are some breaking changes in v6 that don't work with wagmi

> Can it be changed?

We are currently migrating wagmi to use our [viem](https://viem.sh) library (faster, smaller bundle size, more type-safety) instead of ethers v6. More info here: https://github.com/wagmi-dev/wagmi/discussions/1782#discussioncomment-5292169
